packages/flutter/lib/src/widgets/router.dart

```diff
@@ -1311,6 +1317,23 @@ abstract class RouteInformationProvider extends ValueListenable<RouteInformation
   /// The [routeInformation] is the new route information after the navigation
   /// event.
   void routerReportsNewRouteInformation(RouteInformation routeInformation) {}
+
+  /// A callback called when the [Router] widget detects [RouteInformation]
+  /// is updated by a non-navigation event.
```

Comment:
This doc is not super-clear. What is a non-navigation event?

Also: The doc should contrast this to routerReportsNewRouteInformation and point out what the difference is.